На ветке lesson3

RamzanBersanov opened the pull request:
Изменения, которые будут включены в коммит: модели(test, question, user category, answer) и миграции(not null, default value), схема, fixtures

VidgarVii commented on app/models/answer.rb:
```diff
@@ -0,0 +1,3 @@
+class Answer < ApplicationRecord
+  belongs_to :question
```
ассоциации позже будут в курсе
пока лучше убери их из моделей, чтобы они не отвлекали
нам на данном этапе внешних ключей в таблицах вполне достаточно ближайшие пару уроков нужно будет ручками конструировать запросы, чтобы лучше понять, как ассоциации работают, а потом уже будем использовать ассоциации

VidgarVii commented on db/migrate/20221026150404_add_categories_title_null_constraint.rb:
```diff
@@ -0,0 +1,5 @@
+class AddCategoriesTitleNullConstraint < ActiveRecord::Migration[7.0]
+  def change
+    change_column_null(:categories, :title, false)
```
не стоит добавлять отдельные миграции для изменения полей в одном ПР. Миграции всегда являются опасным местом в ПО. Так как меняют схему базы. И всегда стараются минимизировать кол-во миграций.
В этом ДЗ надо создать 5 моделей и 5 таблиц. Значит у тебя должно быть 5 миграций. Прописывай все нужные опции полям сразу. Все остальные миграции являются обычным оверхедом

Пример
```
t.string :title, null: false 
```
Большое кол-во миграций увеличивает время развёртывания ПО. Так же миграция является местом риска, так как может сломать существующую базу. Еще они имеют свойство протухать и быть не актуальными или не рабочими. По этому всегда стараются минимизировать кол-во миграций.

VidgarVii commented on db/schema.rb:
```diff
@@ -0,0 +1,57 @@
+
+ActiveRecord::Schema[7.0].define(version: 2022_10_26_152359) do
+  create_table "answers", force: :cascade do |t|
+    t.text "body", default: "correct", null: false
```
body это текстовое поле, которое должно содержать такст ответа
а correct, который дан по ТЗ, должно быть поле указывающее верный ответ или нет

VidgarVii commented on db/schema.rb:
```diff
@@ -0,0 +1,57 @@
+
+  create_table "users", force: :cascade do |t|
+    t.text "name", null: false
+    t.integer "test_id", null: false
```
юзер не может принадлежать тесту, а вот тесту можно добавить принадлежность к юзеру.
Так как мы будем тесту указывать кто его автор.

RamzanBersanov committed:
На ветке lesson3
 Изменения, которые будут включены в коммит: 5 моделей и 5 миграций для test, answer, category, user, question

RamzanBersanov committed:
На ветке lesson3
 Изменения, которые будут включены в коммит: исправлен аттрибут correct в миграции create_answers, добавлен  author_id в create_tests

RamzanBersanov committed:
Update 20221026181358_create_tests.rb

RamzanBersanov committed:
Update 20221026190730_create_answers.rb
